### DISCUSSION TITLE
Add error to Table.update() without where clause

### DISCUSSION THREAD

**Discussion by theelderbeever** [piccolo-orm/piccolo#409] [Ideas]
As a point of discussion running `await Table.update({"column": "value"})` without a `.where` will cause an update on an entire table. While this is a poor developer mistake to make I wonder if it would be wise to have an arg in the the update to dictate behavior of the ORM when performing such a query. Maybe something like 

```python
>>> await Table.update({"column": "value"}, without_where="error")

UpdateWithoutWhereError("An update on the entire table was attempted please explicitly allow this behavior")
```
Have multiple options such as `error`, `warn`, `ignore`.  And just have it defaulted. Having made this mistake in production it can make for a very bad evening.

Maybe this could exist on `delete` as well?

**Comment by dantownsend** [comment]
@theelderbeever Ah, I'm sorry for the grief this has caused you.

We have something on delete where it won't allow it to run without a where clause, unless you pass `force=True`.

```python
>>> await Band.delete()
Exception

>>> await Band.delete(force=True) # This is allowed
```

Perhaps something similar for update?

I definitely agree something needs adding to handle this scenario.

**Reply by theelderbeever** [reply]
Yeah a `force` arg would definitely do the trick!

P.S. loving how easy piccolo is to work with and extend. You nailed the design pattern

**Reply by dantownsend** [reply]
Thanks!
